x-pack/test/alerting_api_integration/spaces_only/tests/alerting/aggregate.ts

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import expect from '@kbn/expect';
+import { Spaces } from '../../scenarios';
+import { getUrlPrefix, getTestAlertData, ObjectRemover } from '../../../common/lib';
+import { FtrProviderContext } from '../../../common/ftr_provider_context';
+
+// eslint-disable-next-line import/no-default-export
+export default function createAggregateTests({ getService }: FtrProviderContext) {
+  const supertest = getService('supertest');
+
+  describe('aggregate', () => {
+    const objectRemover = new ObjectRemover(supertest);
+
+    afterEach(() => objectRemover.removeAll());
+
+    it('should aggregate when there are no alerts', async () => {
+      const response = await supertest.get(
+        `${getUrlPrefix(Spaces.space1.id)}/api/alerts/_aggregate`
+      );
+
+      expect(response.status).to.eql(200);
+      expect(response.body).to.eql({
+        alertExecutionStatus: {
+          ok: 0,
+          active: 0,
+          error: 0,
+          pending: 0,
+          unknown: 0,
+        },
+      });
+    });
+
+    it('should aggregate alert status totals', async () => {
+      const NumOkAlerts = 4;
+      const NumActiveAlerts = 1;
+      const NumErrorAlerts = 2;
+
+      await Promise.all(
+        [...Array(NumOkAlerts)].map(async () => {
+          const okAlertId = await createTestAlert(
+            {
+              alertTypeId: 'test.noop',
+              schedule: { interval: '1s' },
+            },
+            'ok'
+          );
+          objectRemover.add(Spaces.space1.id, okAlertId, 'alert', 'alerts');
+        })
+      );
+
+      await Promise.all(
+        [...Array(NumActiveAlerts)].map(async () => {
+          const activeAlertId = await createTestAlert(
+            {
+              alertTypeId: 'test.patternFiring',
+              schedule: { interval: '1s' },
+              params: {
+                pattern: { instance: new Array(100).fill(true) },
+              },
+            },
+            'active'
+          );
+          objectRemover.add(Spaces.space1.id, activeAlertId, 'alert', 'alerts');
+        })
+      );
+
+      await Promise.all(
+        [...Array(NumErrorAlerts)].map(async () => {
+          const activeAlertId = await createTestAlert(
+            {
+              alertTypeId: 'test.throw',
+              schedule: { interval: '1s' },
+            },
+            'error'
+          );
+          objectRemover.add(Spaces.space1.id, activeAlertId, 'alert', 'alerts');
+        })
+      );
+
+      await delay(1000);
```

Comment:
Looks like `createTestAlert()` will wait for the alert to hit the desired state, so we shouldn't need to do any additional waiting?  Perhaps this is a hold-over from an earlier version where it didn't wait?

Reply:
chat with @ymao1 , it appears this is probably needed due to ES refresh - `createTestAlert()` is waiting for the status to change, but it may not be available in search until the ES refresh cycle updates search indices.